qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -155,7 +156,9 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
             )
 
         else:
-            return TranspilerError(f"Invalid optimization level {optimization_level}")
+            return TranspilerError(
+                f"Invalid optimization level {optimization_level}"
+            )  # TODO: raise?
```

Comment:
Should this raise error rather than return it?

Reply:
good catch! This is why typechecking is nice. Here is the fix https://github.com/Qiskit/qiskit/pull/12170

Thanks for your work!